src/toast/ToastProvider.tsx

```diff
@@ -6,12 +6,11 @@ import toastReducer from "./util/toastReducer";
 import {ToastAction, ToastContextState} from "./util/toastTypes";
 import {isNonNegativeNumber} from "../core/utils/number/numberUtils";
 
-const ToastContext = createContext<[ToastContextState, React.Dispatch<ToastAction>]>([
-  initialToastState,
-  () => undefined
-]);
+const ToastStateContext = createContext<ToastContextState>(initialToastState);
+const ToastDispatchContext = createContext<React.Dispatch<ToastAction>>(() => undefined);
```

Comment:
Its better to initialize the Context values with `null` or `undefined` so that the Provider is properly set up and the consumers work as expected to avoid unnecessary bugs. As there are custom hooks that we provide for consuming the context value, the nullish cases will be handled there and raise an error if the context is not being used properly.

```suggestion
const ToastDispatchContext = createContext<null | React.Dispatch<ToastAction>>(null);
```

Please update the ToastStateContext as well so that it is initialized with `null`